internal/services/fluidrelay/fluid_relay_servers_resource.go

```diff
@@ -0,0 +1,282 @@
+	//UserAssignedIdentities []UserAssignedIdentity                     `tfschema:"user_assigned_identity"`
+	Identity []identity.ModelSystemAssignedUserAssigned `tfschema:"identity"`
+}
+
+func (s ServerModel) expandUserIdentities() (res *identity.SystemAndUserAssignedMap, err error) {
+	//res := &identity.SystemAndUserAssignedMap{
+	//	Type: "None",
+	//}
+	res, err = identity.ExpandSystemAndUserAssignedMapFromModel(s.Identity)
+	return res, err
+}
+
+func (s *ServerModel) flattenIdentity(input *identity.SystemAndUserAssignedMap) error {
+	if input == nil {
+		return nil
+	}
+	config := identity.SystemOrUserAssignedMap{
+		Type:        input.Type,
+		PrincipalId: input.PrincipalId,
+		TenantId:    input.TenantId,
+		IdentityIds: make(map[string]identity.UserAssignedIdentityDetails),
+	}
+	for k, v := range input.IdentityIds {
+		config.IdentityIds[k] = identity.UserAssignedIdentityDetails{
+			ClientId:    v.ClientId,
+			PrincipalId: v.PrincipalId,
+		}
+	}
+	model, err := identity.FlattenSystemOrUserAssignedMapToModel(&config)
+	if err != nil {
+		return err
+	}
+	s.Identity = *model
+	return nil
+}
+
+type Server struct{}
+
+var _ sdk.ResourceWithUpdate = (*Server)(nil)
+
+func (s Server) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.FluidRelayServerName,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"tags":                commonschema.Tags(),
+		"identity":            commonschema.SystemAssignedUserAssignedIdentityOptional(),
+	}
+}
+
+func (s Server) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"frs_tenant_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"orderer_endpoints": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"storage_endpoints": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"principal_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"tenant_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (s Server) ModelObject() interface{} {
+	return &ServerModel{}
+}
+
+func (s Server) ResourceType() string {
+	return "azurerm_fluid_relay_server"
+}
+
+func (s Server) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.FluidRelay.ServerClient
+
+			var model ServerModel
+			if err = meta.Decode(&model); err != nil {
+				return err
+			}
+
+			account := meta.Client.Account
+			id := fluidrelayservers.NewFluidRelayServerID(account.SubscriptionId, model.ResourceGroup, model.Name)
```

Comment:
Can we change this to:

```suggestion
			subscriptionId := meta.Client.Account.SubscriptionId
			id := fluidrelayservers.NewFluidRelayServerID(subscriptionId, model.ResourceGroup, model.Name)
```